app/javascript/components/util/DeleteDialog.js

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+import { PropTypes } from 'prop-types';
+import { Button, Form, Modal } from 'react-bootstrap';
+
+class DeleteDialog extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      loading: false,
+      disabled: true,
+      showTextInput: false,
+    };
+  }
+
+  handleReasonChange = event => {
+    event.persist();
+    const showTextInput = event.target.value === 'Other';
+    this.setState({ disabled: false, showTextInput, [event.target.id]: event.target.value }, () => {
+      this.props.onChange(event);
+    });
+  };
+
+  handleTextChange = event => {
+    event.persist();
+    this.setState({ [event.target.id]: event.target.value }, () => {
+      this.props.onChange(event);
+    });
+  };
+
+  delete = () => {
+    this.setState({ loading: true }, () => {
+      this.props.delete();
+    });
+  };
+
+  render() {
+    return (
+      <Modal size="lg" show centered onHide={this.props.toggle}>
+        <Modal.Header closeButton>
+          <Modal.Title>Delete {this.props.type}</Modal.Title>
+        </Modal.Header>
+        <Modal.Body>
+          <p>
+            Are you sure you want to delete this {this.props.type}? This action cannot be undone. For auditing purposes, this deletion will be available in this
+            record&apos;s history export.
```

Comment:
I'm wondering if the wording of the last sentence here could be improved. I believe we gernerally use "monitoree" instead of "record" in the UI. And the history entry won't just be available via the export, it will be on the monitoree's profile in their history section. Would something closer to the following be better? Perhaps we should ask the PO team?

> For auditing purposes, this deletion action will create a history entry on this monitoree.

Reply:
@hackrm this is what the PO team gave us, but we can certainly ask

Reply:
Just got this suggested language from the PO team for an updated last sentence:

> For auditing purposes, this deletion will be documented in the record's history.

Reply:
Thanks! -- Addressed.